vllm/v1/core/kv_cache_utils.py

```diff
@@ -231,7 +229,56 @@ def generate_block_hash_extra_keys(
         else:
             # This block has not reached the current mm input.
             break
-    return tuple(extra_keys), curr_mm_idx
+    return extra_keys, curr_mm_idx
+
+
+def generate_block_hash_extra_keys_for_lora(
+        request: Request) -> Optional[List[int]]:
+    """Generate extra keys related to LoRA for block hash computation.
+    
+    Args:
+        request: The request object.
+    
+    Returns:
+        Return LoRA id of the request if it is a LoRA request. Return None
+        otherwise.
+    """
+    if not request.lora_request:
+        return None
+    return [request.lora_request.lora_int_id]
+
+
+def generate_block_hash_extra_keys(
+        request: Request, start_token_idx: int, end_token_idx: int,
+        start_mm_idx: int) -> Tuple[Optional[Tuple[Any, ...]], int]:
+    """Generate extra keys for the block hash. The extra keys can come from
+    the multi-modal inputs and request specific metadata (e.g., LoRA ID).
+    
+    Args:
+        request: The request object.
+        start_token_idx: The start token index of the block.
+        end_token_idx: The end token index of the block.
+        start_mm_idx: The start multi-modal index of the block.
+    
+    Returns:
+        A tuple of extra keys and the next multi-modal index.
+    """
+    mm_extra_keys: Optional[List[Any]]
+    mm_extra_keys, new_start_mm_idx = generate_block_hash_extra_keys_for_mm(
+        request, start_token_idx, end_token_idx, start_mm_idx)
+    lora_extra_keys: Optional[
+        List[int]] = generate_block_hash_extra_keys_for_lora(request)
+
+    extra_keys: List[Any] = []
+    if mm_extra_keys:
+        extra_keys.extend(mm_extra_keys)
+    if lora_extra_keys:
+        extra_keys.extend(lora_extra_keys)
+
+    if not extra_keys:
+        return None, new_start_mm_idx
+
+    return tuple(extra_keys), new_start_mm_idx
```

Comment:
I think we can unify the extra key format to remove `None` (just return `[]` instead), so that this logic can be simplified:

```python
    extra_keys: List[Any] = mm_extra_keys + lora_extra_keys
    return tuple(extra_keys), new_start_mm_idx
```